Run golint by default in travis ci

chentex opened the pull request:
**What this PR does / why we need it**:
Adds golint target in GNUmakefile
Installs golint dependency in bootstrap target
Adds before script in travis.yml

**Which issue this PR fixes** : fixes #78 

**Special notes for your reviewer**:
Also fixed golint suggestions to have it clean from now onwards.

prateekpandey14 commented on GNUmakefile:
```diff
@@ -64,6 +65,9 @@ format:
 	@go fmt $(PACKAGES)
 
+golint:
+	golint
```
This tag will only run `golint` tool in root folder , this issue is for running `golint` in all pkg's inside the openebs/maya repo.

prateekpandey14 replied:
It will be better to have `gometalinter`  which will be aggregate and normalise the output of a whole bunch of Go linters, not just `golint`

AmitKumarDas commented on GNUmakefile:
```diff
@@ -64,6 +65,9 @@ format:
 	@go fmt $(PACKAGES)
 
+golint:
+	golint
```
Did you check by running @golint $(PACKAGES)

chentex replied:
I didn't yet, I'll work on this changes tonight. I did test the way I have and it's only running on the root.

chentex commented:
**UPDATE**
Started using gometalinter, some point I'll clarify:

`-gometalinter.v1 --vendor --deadline=600s ./...`

**1.** I had to use `./...` because using `$(PACKAGES)` used the full URL and gometalint didn't find anything to test.
**2.** Had to add `--deadline=600s` because it takes a lot of time for the lits to pass, otherwise it didn't finish
**3.** Had to use **`-`** to ignore exit code, so the build could continue, there are tons of warnings and errors that have to be checked.
**4.** I think doing fixes at this point will only complicate this PR more than it needs to be; I would prefer to create separate PR for every 2 or 3 different linters that gometalinter uses.

Please share your comments and concerns @prateekpandey14 @AmitKumarDas

kmova commented:
I agree with @chentex -- Keeping this PR open and having more fixes checked in, will complicate merging this PR.

prateekpandey14 commented on .travis.yml:
```diff
@@ -8,6 +8,9 @@ go:
 install:
   - make bootstrap
+before_script:
+  - make deps
```
`make deps` required here ?

chentex replied:
I had to include it here, because cycle lint was failing because of missing deps.

AmitKumarDas commented on .travis.yml:
```diff
@@ -8,6 +8,9 @@ go:
   - make bootstrap
+before_script:
+  - make deps
+  - make golint
```
Both make deps & make golint are timing out the travis run.
Let us remove make deps
& check if make golint can run only for maya source code. It should not run for vendor/ code

prateekpandey14 replied:
`make dep` is not timing out ( target  is `dep ensure` which do not enable for verbose logging mode)
if we do `dep ensure -v` it will show the verbose logging output.

chentex replied:
It's `make golint` the one that is timing out, I added a 10 minutes timeout for that, and in travis it's taking longer than that.

Maybe reducing the linters that are validated in travis and only make it full for local environment?

kmova replied:
sounds good @chentex -- running full on local is a good idea.

chentex commented:
Hello, so I just pushed more changes:
1. Added a new target for lint in Travis, using a config file will be easier to config what to run.
2. Leaved the later target for devs to run it locally.
3. Updated the developer.md to make devs run the linter locally, so hopefully, altogether we can fix everything.

Let me know your thoughts.

prateekpandey14 commented on Dockerfile:
```diff
@@ -0,0 +1,27 @@
+#
```
Remove this Dockerfile, its already merged with other commit.

prateekpandey14 commented on buildscripts/apiserver/Dockerfile:
```diff
@@ -3,17 +3,18 @@
 # maya api server's releases.
 #
 
```
Remove this file too.

prateekpandey14 commented on buildscripts/mayactl/Dockerfile:
```diff
@@ -2,7 +2,9 @@
 # This Dockerfile builds a recent maya container with latest code
 #
 
```
remove this too.

prateekpandey14 commented:
@chentex thanks again, 
Needs to do the following : 
1. Remove the all `Dockerfiles` which are part of PR( mentioned is review comments)
2. Squash all the commits into 1 commit or so.

chentex commented:
@prateekpandey14 just pushed the changes, removed the dockerfiles and sqashed it to three commits.
